tests/bluetooth/host/mock_host/src/net.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include <stddef.h>
+#include <string.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/net/buf.h>
+#include <zephyr/bluetooth/buf.h>
+
+
+static uint8_t *fixed_data_alloc(struct net_buf *buf, size_t *size,
+			      k_timeout_t timeout)
+{
+	ztest_test_fail();
+	return NULL;
+}
+
+static void fixed_data_unref(struct net_buf *buf, uint8_t *data)
+{
+	/* Nothing needed for fixed-size data pools */
+	ztest_test_fail();
+}
+
+const struct net_buf_data_cb net_buf_fixed_cb = {
+	.alloc = fixed_data_alloc,
+	.unref = fixed_data_unref,
+};
+
+struct net_buf *net_buf_alloc_fixed(struct net_buf_pool *pool, k_timeout_t timeout)
+{
+	zassert_not_null(pool, "Value was NULL");
```

Comment:
redundant message. Changing it to "Invalid pool" would be better (or even removing it is fine).

Reply:
I also think this check might be un-necessary, seeing how you're checking its value anyways in the next line. So the test case function should take care of checking if `pool` is `NULL` or not IMO.

Reply:
Correct, It's a redundant check.
I think it was used as a test in the beginning but it was left over.